massgov/mayflower
# React/DP-10868  a11y tabs container focus group

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

Set up keyboard navigation between the tabs and their associated tab content with arrow keys.

In a tab group:
1. LEFT/RIGHT ARROW keys to navigate in the tabs within either the parent group or the child (= nested) group.
2. DOWN ARROW or ENTER key to navigate from the active tab to its tab content. (note: the focus is on the tab content container, not the first focusable element in the container.)

In a tab content container:
3. UP ARROW key or shift+TAB to move the focus from the tab content container to the active tab.
In tab content container, TAB key moves the keyboard focus to the closest focusable element in the container. 

Arrow keys are used only to navigate between the tabs and the active tab and its content container.

## Related Issue / Ticket

- [DP-10868](https://jira.mass.gov/browse/DP-10868)

## Steps to Test


1. Go to tab container in organisms.
2. Use LEFT/RIGHT ARROW keys to navigate between the tabs.
     - No errors at the first and last tabs by trying to navigate to non-existing tabs. It stays at the current tab.
     - The current keyboard focus matches the active tab except for when the component is rendered first time.
3. Use DOWN ARROW or ENTER key to navigate to the active tab's content area.
4. Use UP ARROW or Shift+Tab to navigate from the content area to its associate tab (= current active tab).

- The keyboard focus stays with the active tab when navigating in the tab group.
- When navigating to the tab content container from the active tab, the keyboard focus moves to the tab content container.  You see it as blue outline on the container as it gets a keyboard focus.
- The visual indicator of the active tab remains while its content area or contents of the content area has the keyboard focus. 

Navigation patterns between the active tab and its tab content:
1. active tab -> tab container -> element in tab container -> ... -> tab container -> active tab
2. parent active tab -> parent tab container -> child(= nested) active tab -> child tab container -> ... -> child(= nested) active tab -> parent tab container -> parent active tab

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
